scripts/contributor_dashboard_debug.py

```diff
@@ -0,0 +1,265 @@
+        generated automatically from email.
+        """
+        # In development mode, the "Sign In" button will call the function
+        # AuthService.signInWithEmail() in the file
+        # core/templates/services/auth.service.ts. That function generates a
+        # password through the MD5 hash of email addresses to save developers
+        # and end-to-end test authors the trouble of providing passwords. This
+        # script also uses the MD5 hash when creating a user in the Firebase
+        # Authentication emulator. When we want to sign in a user created using
+        # this script, clicking the "Sign In" button will generate a valid
+        # password which is the same as that in the emulator.
+        password = hashlib.md5(email.encode('utf-8')).hexdigest()
+
+        # The FIREBASE_AUTH_EMULATOR_HOST environment variable is set to connect
+        # with the Firebase Authentication emulator.
+        with self._set_environ(
+            'FIREBASE_AUTH_EMULATOR_HOST', FIREBASE_AUTH_EMULATOR_HOST
+        ):
+            # Create a new user in Firebase.
+            firebase_auth.create_user(email=email, password=password)
+
+        # Sign up the new user in Oppia and set its username.
+        self._sign_in(email)
+        self._make_request('GET', '/signup?return_url=/')
+        self.csrf_token = self._get_csrf_token()
+
+        params = {'payload': json.dumps({
+            'username': username,
+            'agreed_to_terms': True,
+            'default_dashboard': constants.DASHBOARD_TYPE_LEARNER,
+        }), 'csrf_token': self.csrf_token}
+
+        self._make_request('POST', feconf.SIGNUP_DATA_URL, params=params)
+
+        # End current session, i.e. log out.
+        self._make_request('GET', '/session_end')
+        self.csrf_token = ''
+
+    @contextlib.contextmanager
+    def _set_environ(self, env_name: str, value: str) -> Iterator[None]:
+        """Temporarily set the environment variable 'env_name' to 'value' while
+        inside the ``with`` block.
+        """
+        old_value = os.environ.get(env_name)
+        os.environ[env_name] = value
+        try:
+            yield
+        finally:
+            if old_value is None:
+                del os.environ[env_name]
+            else:
+                os.environ[env_name] = old_value
+
+    def _sign_in(self, email: str) -> None:
+        """Begins a session with the given email, i.e. log in with the email."""
+        password = hashlib.md5(email.encode('utf-8')).hexdigest()
+
+        token_id = self._sign_in_with_email_and_password(email, password)
+        headers = {'Authorization': 'Bearer %s' % token_id}
+
+        self._make_request('GET', '/session_begin', headers=headers)
+
+    def _sign_in_with_email_and_password(
+        self, email: str, password: str
+    ) -> str:
+        """Signs in with email and password, and returns the token id."""
+        token_id = str(requests.post(
```

Comment:
is `idToken` always a string, or is none possible?

If `None` is possible, conversion to string won't give you what is expected